src/main/java/core/basesyntax/Lottery.java

```diff
@@ -0,0 +1,16 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class Lottery extends ColorSupplier {
```

Comment:
- There is no need to extend ColorSupplier

- Avoid adding empty lines, if there is no logical need to divide code blocks

- ColorSupplier getRandomColor() class is no longer static, so it should be called by its instance. Additionally, it shouldn't be a local method variable, you should move ColorSupplier initialization to the class-level.

- As explained in the ColorSupplier class, you should avoid creating a Random object each time the method is called. The best approach is to initialize one random object at the class-level and reuse it as a constructor parameter for ColorSupplier object and to generate a random ball number.

- You've created all arguments constructor for Ball object, you should use it for better readability (with that change you can get rid of setters in Ball class).

- you should move the bounding number of nextInt method do the static constant variable.

```suggestion
package core.basesyntax;

import java.util.Random;

public class Lottery {
  // create static final constant variable for the maximum value of the ball number
  public static final int BALL_MAX_NUMBER = 100;
  // Initialize random and colorSupplierObjects on the class-level
  Random random = new Random();
  // Use random object as a constructor parameter
  ColorSupplier colorSupplier = new ColorSupplier(random);

  public Ball getRandomBall() {
    String ballColor = colorSupplier.getRandomColor();
    int ballNumber = random.nextInt(BALL_MAX_NUMBER);
    // Create and directly return ball object using all arguments constructor
    return new Ball(ballColor, ballNumber);
  }
}

```